game_patch/object/entity.cpp

```diff
@@ -201,6 +202,79 @@ CodeInjection entity_process_pre_hide_riot_shield_injection{
     },
 };
 
+FunHook<void(int)> entity_blood_throw_gibs_hook{
+    0x0042E3C0, [](int handle) {
+        if (g_game_config.gibbing && !rf::is_multi) {
+            rf::Object* objp = rf::obj_from_handle(handle);
+
+            if (!objp) {
+                return; // invalid object
+            }
+
+            rf::Vector3 pos = objp->pos;
+            rf::GRoom* room = objp->room;
+            int explosion_vclip = 30;
```

Comment:
Good point. I've added the `vclip_lookup` method and hardcoded a vclip name rather than index. It's still just setting a default though, mod developers can override it by specifying an `$Explode Anim` in the correspondign `entity.tbl` class.